File: food-donation-frontend/src/services/dashboardService.js
import apiClient from './apiClient';

const dashboardService = {
  async getDonorDashboard() {
    const { data } = await apiClient.get('/dashboard/donor');
    return data.data;
  },
  async getNgoDashboard() {
    const { data } = await apiClient.get('/dashboard/ngo');
    return data.data;
  },
  async getAgentDashboard() {
    const { data } = await apiClient.get('/dashboard/agent');
    return data.data;
  },
};

export default dashboardService;

File: food-donation-backend/src/main/java/com/fooddonation/service/DashboardService.java
package com.fooddonation.service;

import com.fooddonation.dto.response.DashboardStatsResponse;

public interface DashboardService {

	DashboardStatsResponse getAdminDashboard();

	DashboardStatsResponse getDonorDashboard(String donorEmail);

	DashboardStatsResponse getNgoDashboard(String ngoEmail);

	DashboardStatsResponse getAgentDashboard(String agentEmail);

}
